tests/app/utilities/test_decimal_places.py

```diff
@@ -0,0 +1,129 @@
+            "es_ES",
+            "short",
+            "0,123 mm",
+        ),
+        (123, "centimeter", "en_GB", "short", "123 cm"),
+        (123, "centimeter", "pl_PL", "short", "123 cm"),
+        (123, "kilometer", "en_GB", "long", "123 kilometres"),
+        (3, "kilometer", "en_GB", "long", "3 kilometres"),
+        (3, "kilometer", "hu_HU", "long", "3 kilométer"),
+        (Decimal("1.2"), "kilometer", "en_GB", "long", "1.2 kilometres"),
+        (Decimal("1.20"), "kilometer", "en_GB", "long", "1.20 kilometres"),
+        (Decimal("1.200"), "kilometer", "en_GB", "long", "1.200 kilometres"),
+        (Decimal("1.2000"), "kilometer", "en_GB", "long", "1.2000 kilometres"),
+        (Decimal("1.20000"), "kilometer", "en_GB", "long", "1.20000 kilometres"),
+        (Decimal("1.20000"), "kilometer", "de_DE", "long", "1,20000 Kilometer"),
+        (Decimal("1.2"), "kilometer", "pl_PL", "long", "1,2 kilometra"),
+        (Decimal("1.2345"), "kilometer", "en_GB", "long", "1.2345 kilometres"),
+        (Decimal("1.2345"), "kilometer", "es_ES", "long", "1,2345 kilómetros"),
+        (123, "mile", "en_GB", "short", "123 mi"),
+        (123, "mile", "en_GB", "narrow", "123mi"),
+        (123, "mile", "en_US", "narrow", "123mi"),
+    ),
+)
+def test_custom_format_unit(
+    value, measurement_unit, locale_string, length, expected_result
+):
+    result = custom_format_unit(value, measurement_unit, locale_string, length)
+
+    assert result == expected_result
+
+
+@pytest.mark.parametrize(
+    "value, currency, locale_string, decimal_limit, expected_result",
+    (
+        (123, "GBP", "en_GB", 1, "£123"),
+        (Decimal("123.1234"), "GBP", "en_GB", 0, "£123"),
+        (Decimal("3000.44545"), "GBP", "en_GB", None, "£3,000"),
+        (Decimal("2.1"), "GBP", "en_GB", None, "£2"),
+        (Decimal("3000"), "GBP", "en_GB", 0, "£3,000"),
+        (Decimal("3000"), "JPY", "en_GB", 0, "JP¥3,000"),
+        (Decimal("3000"), "JPY", "ja_JP", 0, "¥3,000"),
+        (Decimal("123.45"), "GBP", "en_GB", 1, "£123.4"),
+        (Decimal("123.45"), "HUF", "hu_HU", 1, "123,4 Ft"),
+        (Decimal("2.1"), "GBP", "en_GB", 1, "£2.1"),
+        (11, "GBP", "en_GB", 2, "£11.00"),
+        (11000, "USD", "en_GB", 2, "US$11,000.00"),
+        (11000, "USD", "en_GB", 2, "US$11,000.00"),
+        (11000, "PLN", "pl_PL", 2, "11 000,00 zł"),
+        (Decimal("11.99"), "GBP", "en_GB", 2, "£11.99"),
+        (Decimal("2.1"), "GBP", "en_GB", 4, "£2.10"),
+        (Decimal("2.1"), "GBP", "en_GB", 5, "£2.10"),
+        (2, "GBP", "en_GB", 6, "£2.00"),
+        (Decimal("2.14564"), "GBP", "en_GB", 6, "£2.14564"),
+        (Decimal("1.1"), "GBP", "en_GB", 6, "£1.10"),
+        (Decimal("1.10"), "GBP", "en_GB", 6, "£1.10"),
+        (Decimal("1.100"), "GBP", "en_GB", 6, "£1.100"),
+        (Decimal("1.1000"), "GBP", "en_GB", 6, "£1.1000"),
+        # (Decimal("3000.445"), "GBP", "en_GB", 6, "£3,000.445"),
+    ),
+)
+def test_custom_format_currency(
```

Comment:
Same as above, why do we have this and `TEST_FORMAT_CURRENCY_PARAMS`. If you handle the locale you have a single test set for both?.

Reply:
Updated 👍